src/Microsoft.Crank.Jobs.K6/Program.cs

```diff
@@ -0,0 +1,367 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Crank.EventSources;
+
+namespace Microsoft.Crank.Jobs.K6
+{
+    static class Program
+    {
+        private static readonly HttpClient _httpClient;
+        private static readonly HttpClientHandler _httpClientHandler;
+
+        static Program()
+        {
+            // Configuring the http client to trust the self-signed certificate
+            _httpClientHandler = new HttpClientHandler();
+            _httpClientHandler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
+            _httpClientHandler.AutomaticDecompression = System.Net.DecompressionMethods.GZip | System.Net.DecompressionMethods.Deflate;
+
+            _httpClient = new HttpClient(_httpClientHandler);
+        }
+
+        static async Task<int> Main(string[] args)
+        {
+            Console.WriteLine("K6 Client");
+            Console.WriteLine("args: " + String.Join(' ', args));
+
+            Console.Write("Measuring first request ... ");
+            await MeasureFirstRequest(args);
+
+            // Extracting parameters
+            var argsList = args.ToList();
+
+            TryGetArgumentValue("--warmup", argsList, out int warmup);
+            TryGetArgumentValue("--duration", argsList, out int duration);
+
+            args = argsList.ToArray();
+
+            string k6Url = null;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                switch (RuntimeInformation.ProcessArchitecture)
+                {
+                    case Architecture.X64 : k6Url = "https://aspnetbenchmarks.blob.core.windows.net/tools/k6-win-amd64.exe"; break;
```

Comment:
no equivalent for win-arm64?

Reply:
I would have to build it on these machines, and this is not a scenario we are looking at.